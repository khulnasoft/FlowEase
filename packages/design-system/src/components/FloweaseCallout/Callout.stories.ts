import FloweaseCallout from './Callout.vue';
import FloweaseLink from '../FloweaseLink';
import FloweaseText from '../FloweaseText';
import type { StoryFn } from '@storybook/vue3';

export default {
	title: 'Atoms/Callout',
	component: FloweaseCallout,
	argTypes: {
		theme: {
			control: {
				type: 'select',
			},
			options: ['info', 'secondary', 'success', 'warning', 'danger', 'custom'],
		},
		message: {
			control: {
				type: 'text',
			},
		},
		icon: {
			control: {
				type: 'text',
			},
		},
	},
	parameters: {
		design: {
			type: 'figma',
			url: 'https://www.figma.com/file/tPpJvbrnHbP8C496cYuwyW/Node-pinning?node-id=15%3A5777',
		},
	},
};

interface Args {
	theme: string;
	icon: string;
	default: string;
	actions: string;
	trailingContent: string;
}

const template: StoryFn<Args> = (args, { argTypes }) => ({
	setup: () => ({ args }),
	props: Object.keys(argTypes),
	components: {
		FloweaseLink,
		FloweaseText,
		FloweaseCallout,
	},
	template: `
		<flowease-callout v-bind="args">
			${args.default}
			<template #actions v-if="args.actions">
				${args.actions}
			</template>
			<template #trailingContent v-if="args.trailingContent">
				${args.trailingContent}
			</template>
		</flowease-callout>
	`,
});

export const defaultCallout = template.bind({});
defaultCallout.args = {
	theme: 'success',
	default: `
		This is a default callout.
	`,
};

export const customCallout = template.bind({});
customCallout.args = {
	theme: 'custom',
	icon: 'code-branch',
	default: `
		This is a custom callout.
	`,
	actions: `
		<flowease-link size="small">
			Do something!
		</flowease-link>
	`,
};

export const secondaryCallout = template.bind({});
secondaryCallout.args = {
	theme: 'secondary',
	icon: 'thumbtack',
	default: `
		This data is pinned.
	`,
	actions: `
		<flowease-link theme="secondary" size="small" :bold="true" :underline="true">
			Unpin
		</flowease-link>
	`,
	trailingContent: `
		<flowease-link
			theme="secondary"
			size="small"
			:bold="true"
			:underline="true"
			to="https://flowease.khulnasoft.com"
		>
			Learn more
		</flowease-link>
	`,
};
