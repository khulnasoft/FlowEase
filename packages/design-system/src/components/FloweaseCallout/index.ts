import FloweaseCallout from './Callout.vue';
export type { CalloutTheme } from './Callout.vue';
export default FloweaseCallout;
