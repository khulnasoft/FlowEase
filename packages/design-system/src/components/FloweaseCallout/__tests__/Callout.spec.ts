import { render } from '@testing-library/vue';
import FloweaseCallout from '../Callout.vue';

describe('components', () => {
	describe('FloweaseCallout', () => {
		it('should render info theme correctly', () => {
			const wrapper = render(FloweaseCallout, {
				props: {
					theme: 'info',
				},
				global: {
					stubs: ['flowease-icon', 'flowease-text'],
				},
				slots: {
					default: '<flowease-text size="small">This is an info callout.</flowease-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render success theme correctly', () => {
			const wrapper = render(FloweaseCallout, {
				props: {
					theme: 'success',
				},
				global: {
					stubs: ['flowease-icon', 'flowease-text'],
				},
				slots: {
					default: '<flowease-text size="small">This is a success callout.</flowease-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render warning theme correctly', () => {
			const wrapper = render(FloweaseCallout, {
				props: {
					theme: 'warning',
				},
				global: {
					stubs: ['flowease-icon', 'flowease-text'],
				},
				slots: {
					default: '<flowease-text size="small">This is a warning callout.</flowease-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render danger theme correctly', () => {
			const wrapper = render(FloweaseCallout, {
				props: {
					theme: 'danger',
				},
				global: {
					stubs: ['flowease-icon', 'flowease-text'],
				},
				slots: {
					default: '<flowease-text size="small">This is a danger callout.</flowease-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render secondary theme correctly', () => {
			const wrapper = render(FloweaseCallout, {
				props: {
					theme: 'secondary',
				},
				global: {
					stubs: ['flowease-icon', 'flowease-text'],
				},
				slots: {
					default: '<flowease-text size="small">This is a secondary callout.</flowease-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render custom theme correctly', () => {
			const wrapper = render(FloweaseCallout, {
				props: {
					theme: 'custom',
					icon: 'code-branch',
				},
				global: {
					stubs: ['flowease-icon', 'flowease-text'],
				},
				slots: {
					default: '<flowease-text size="small">This is a secondary callout.</flowease-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render additional slots correctly', () => {
			const wrapper = render(FloweaseCallout, {
				props: {
					theme: 'custom',
					icon: 'code-branch',
				},
				global: {
					stubs: ['flowease-icon', 'flowease-text', 'flowease-link'],
				},
				slots: {
					default: '<flowease-text size="small">This is a secondary callout.</flowease-text>',
					actions: '<flowease-link size="small">Do something!</flowease-link>',
					trailingContent:
						'<flowease-link theme="secondary" size="small" :bold="true" :underline="true" to="https://flowease.khulnasoft.com">Learn more</flowease-link>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
