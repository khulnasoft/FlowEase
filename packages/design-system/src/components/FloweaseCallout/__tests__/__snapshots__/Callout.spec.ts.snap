// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > FloweaseCallout > should render additional slots correctly 1`] = `
"<div class="flowease-callout callout custom round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <flowease-icon-stub icon="code-branch" size="medium" spin="false"></flowease-icon-stub>
    </div>
    <flowease-text-stub bold="false" size="small" compact="false" tag="span"></flowease-text-stub> &nbsp; <flowease-link-stub size="small"></flowease-link-stub>
  </div>
  <flowease-link-stub theme="secondary" size="small" bold="true" underline="true" to="https://flowease.khulnasoft.com"></flowease-link-stub>
</div>"
`;

exports[`components > FloweaseCallout > should render custom theme correctly 1`] = `
"<div class="flowease-callout callout custom round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <flowease-icon-stub icon="code-branch" size="medium" spin="false"></flowease-icon-stub>
    </div>
    <flowease-text-stub bold="false" size="small" compact="false" tag="span"></flowease-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > FloweaseCallout > should render danger theme correctly 1`] = `
"<div class="flowease-callout callout danger round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <flowease-icon-stub icon="exclamation-triangle" size="medium" spin="false"></flowease-icon-stub>
    </div>
    <flowease-text-stub bold="false" size="small" compact="false" tag="span"></flowease-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > FloweaseCallout > should render info theme correctly 1`] = `
"<div class="flowease-callout callout info round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <flowease-icon-stub icon="info-circle" size="medium" spin="false"></flowease-icon-stub>
    </div>
    <flowease-text-stub bold="false" size="small" compact="false" tag="span"></flowease-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > FloweaseCallout > should render secondary theme correctly 1`] = `
"<div class="flowease-callout callout secondary round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <flowease-icon-stub icon="info-circle" size="medium" spin="false"></flowease-icon-stub>
    </div>
    <flowease-text-stub bold="false" size="small" compact="false" tag="span"></flowease-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > FloweaseCallout > should render success theme correctly 1`] = `
"<div class="flowease-callout callout success round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <flowease-icon-stub icon="check-circle" size="medium" spin="false"></flowease-icon-stub>
    </div>
    <flowease-text-stub bold="false" size="small" compact="false" tag="span"></flowease-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > FloweaseCallout > should render warning theme correctly 1`] = `
"<div class="flowease-callout callout warning round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <flowease-icon-stub icon="exclamation-triangle" size="medium" spin="false"></flowease-icon-stub>
    </div>
    <flowease-text-stub bold="false" size="small" compact="false" tag="span"></flowease-text-stub> &nbsp;
  </div>
</div>"
`;
